.github/workflows/ci.yml

```diff
@@ -0,0 +1,61 @@
+name: Build mobile-apps-article-templates
+
+on:
+    pull_request:
+        branches:
+            - master
+    push:
+        branches:
+            - master
+    workflow_dispatch:
+        inputs:
+            manual_trigger:
+                description: "Manually trigger the workflow"
+                required: false
+
+jobs:
+    CI:
+        if: github.actor != 'dependabot[bot]'
+        permissions:
+            id-token: write
+            contents: read
+        name: mobile-apps-article-templates build
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v2
+
+            - name: Read Node version from .nvmrc
+              id: read-node-version
+              run: echo "::set-output name=node_version::$(cat .nvmrc)"
+
+            - name: Setup Node
+              uses: actions/setup-node@v2
+              with:
+                  node-version: ${{ steps.read-node-version.outputs.node_version }}
+                  cache: "npm"
+
+            - name: Install dependencies
+              run: npm install
+
+            - name: Build and test
+              run: |
+                  npm run build
+                  npm run test
+
+            - name: Set up npm auth token if not Dependabot
+              run: |
+                  if [[ ${{ github.actor }} != 'dependabot[bot]' ]]; then
+                      echo "//registry.npmjs.org/:_authToken=${NODE_TOKEN}" >> ~/.npmrc
+                  fi
+              env:
+                  NODE_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+            - name: Bump version and publish if not Dependabot
+              if: success() && github.actor != 'dependabot[bot]'
+              run: |
+                  OFFSET=858
+                  PATCH_VERSION=$((OFFSET + ${{ github.run_number }}))
+
+                  npm --no-git-tag-version version 1.0.${PATCH_VERSION}
+                  npm publish --access public
```

Comment:
I think this is neat - we can choose not to publish to npm if the script is being executed by a dependabot pr,  and we can also not bump the workflow if the workflow is unsuccessful which should ensure no gaps in our versions if the workflow is unsuccessful. Let me know if you think this is a good idea!